### PR TITLE
aaron linting for PermissionsManagement component

### PR DESCRIPTION
**Pull request by AaronPersaud**
# Description
Fix linting for the `PermissionsManagement` component

## Related PRS (if any):
N/A

## Main changes explained:
- removed the PermissionsManagement component directory from `.eslintignore` and `.prettierignore`

## How to test:
1. check into current branch
2. do `npm install` and `npm run lint` to make sure you get no lint errors
3. run `npm run start:local` and start the backend as usual
4. go to Other Links -> Permissions Management and make sure everything is working (please do not add/remove any permissions).
5. run `npm run test` to make sure the tests are passing (see below image)

image

## Note:
- the failing tests are also on dev branch, I am looking into that
- there were no lint errors, so I haven't really changed any of the component's code so nothing should be breaking

### CODE REVIEW

**Comment by Arsamat** [comment]
Hello @AaronPersaud I tested your code and the changes seem to be working correctly.
Снимок экрана (256)
Снимок экрана (257)
Снимок экрана (254)

**Review by Arsamat — APPROVED** [review]
Tested and changes worked as expected

**Review by Haoj1 — DISMISSED** [review]
@AaronPersaud Good job! All tests passed and no lint error

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/ae362466-29e9-483c-8ca0-f565973d5441

**Review by sxiong5 — CHANGES_REQUESTED** [review]
Hey, `npm run test` doesn't use eslint and prettier to check the code.
All the `jsx` file have issues that need to be solved.
Please check them one by one to solve it.

**Tips: `npx eslint --fix [path]` can help solve the issues automatically.**
Screenshot 2023-08-28 at 15 15 42

**Comment by AaronPersaud** [comment]
Hey @sxiong5, thanks for reviewing!

I'm not receiving those errors on my VS Code.

image

I'm guessing it's because I'm missing a plugin? I have prettier plugin installed, do I need eslint too? I assumed that all the eslint would be checked with `npm run lint`. Please let me know. Thank you.

image

**Comment by sxiong5** [comment]
> Hey @sxiong5, thanks for reviewing!
> 
> I'm not receiving those errors on my VS Code.
> 
> image
> 
> I'm guessing it's because I'm missing a plugin? I have prettier plugin installed, do I need eslint too? I assumed that all the eslint would be checked with `npm run lint`. Please let me know. Thank you.
> 
> image

Hey @AaronPersaud, you still need ESLint in your vscode. And the command `npm run lint` will only check the `index.js` on the folder if that exists. Otherwise, it won't display any errors. So what you need to do is use `npx lint [path]` to check the file one by one.

**Review by scoopstroop92 — COMMENTED** [review]
Hello @AaronPersaud  Good work on the fix . However I still see the following warnings. 
Screenshot (4)
Screenshot (5)

Screenshot (6)
Screenshot (7)

**Comment by AaronPersaud** [comment]
@ramyaram2092, thanks for reviewing. However, I'm not getting the errors that you're getting.

image

I'm going to further look into this, I'm not really sure why there's a discrepancy. If you have any ideas, feel free to let me know.

**Comment by AaronPersaud** [comment]
@ramyaram2092 I fixed all the errors, I realized there's something wrong with my VS Code and some of these errors don't show up. As for the errors in the test file, the linter is setup to ignore those.

image

Please see if all the errors have cleared for you. Thank you.

**Review by scoopstroop92 — COMMENTED** [review]
Hello @AaronPersaud now i dont see the issues I mentioned before .
however I just see "unable to resolve path" warning in all the files.  if that's fixed we are good to go 


Screenshot (12)

**Review by EvianTan — CHANGES_REQUESTED** [review]

File: src/components/PermissionsManagement/NewRolePopUp.jsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
-import React, { useState } from 'react';
+import { useState } from 'react';
 import { FormCheck } from 'react-bootstrap';
 import { Alert, Button, Form, FormGroup, Input, Label } from 'reactstrap';
 import { toast } from 'react-toastify';
 import { connect } from 'react-redux';
 import { boxStyle } from 'styles';
 import { addNewRole, getAllRoles } from '../../actions/role';
-import { permissionLabel } from './UserRoleTab';
+import { permissionLabel } from './RolePermissions';
```
[the remark this answers is not in the document]
Comment (AaronPersaud) [reply src/components/PermissionsManagement/NewRolePopUp.jsx:8]: So before my change, `UserRoleTab` imports the entire component `RolePermissions` and then `RolePermissions` was importing the `permissionLabel` constant from `UserRoleTab` which was causing a circular dependency. I moved `permissionLabel` to `RolePermissions` so then `RolePermissions` doesn't have to import it from `UserRoleTab`, and it breaks the circle.

**Comment by EvianTan** [comment]
I'm taking over this PR and I'll work on it soon. Need to pull in the latest changes from development and fix all the eslint/prettier issues. And also make sure the builds won't fail.

**Comment by EvianTan** [comment]
Closed this PR for now because I raised another PR #2269 instead.

**Pull request closed without merging**